ogr/ogrsf_frmts/miramon/mm_rdlayr.c

```diff
@@ -0,0 +1,674 @@
+ *           mainly written by Xavier Pons, Joan Maso (correctly written
+ *           "Mas0xF3"), Abel Pau, Nuria Julia (N0xFAria Juli0xE0),
+ *           Xavier Calaf, Lluis (Llu0xEDs) Pesquer and Alaitz Zabala, from
+ *           CREAF and Universitat Autonoma (Aut0xF2noma) de Barcelona.
+ *           For a complete list of contributors:
+ *           https://www.miramon.cat/eng/QuiSom.htm
+ ******************************************************************************
+ * Copyright (c) 2024, Xavier Pons
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE. 
+ ****************************************************************************/
+
+#ifdef GDAL_COMPILATION
+#include "ogr_api.h"  // For CPL_C_START
+#include "mm_wrlayr.h"
+#include "mm_wrlayr.h"  // For MMReadHeader()
+#include "mm_gdal_functions.h"
+#include "mm_gdal_constants.h"
+#else
+#include "CmptCmp.h"                    // Compatibility between compilers
+#include "mm_gdal\mm_wrlayr.h"          // For MMReadHeader()
+#include "mm_gdal\mm_gdal_functions.h"  // For int MM_GetArcHeights()
+#include "mm_constants.h"
+#endif
+
+#include "mm_rdlayr.h"
+
+#ifdef GDAL_COMPILATION
+CPL_C_START  // Necessary for compiling in GDAL project
+#endif
+
+    static char local_message[MAX_LOCAL_MESSAGE];
+
+/* -------------------------------------------------------------------- */
+/*      Reading MiraMon format file functions                           */
+/* -------------------------------------------------------------------- */
+
+// Initializes a MiraMon vector layer for reading
+int MMInitLayerToRead(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                      FILE_TYPE *m_fp, const char *pszFilename)
+{
+    char szResult[MM_MAX_ID_SNY + 10];
+    char *pszSRS;
+
+    memset(hMiraMonLayer, 0, sizeof(*hMiraMonLayer));
+    if (MMReadHeader(m_fp, &hMiraMonLayer->TopHeader))
+    {
+        sprintf(local_message, "Error reading header of file %s", pszFilename);
```

Comment:
snprintf(local_message, sizeof(local_message), ...) to be used everywhere sprintf(local_message, ...) is used